server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainIT.java

```diff
@@ -66,12 +72,19 @@
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
 public final class ClusterAllocationExplainIT extends ESIntegTestCase {
 
-    public void testUnassignedPrimaryWithExistingIndex() throws Exception {
+    public void testUnassignedPrimaryWithExistingOpenedIndex() throws Exception {
+        runTestUnassignedPrimaryWithExistingIndex(IndexMetaData.State.OPEN);
+    }
+
+    public void testUnassignedPrimaryWithExistingClosedIndex() throws Exception {
```

Comment:
I don't think it makes sense to duplicate these tests, given that closed indices will now behave exactly the same way as open indices when it comes to allocation. Just adding randomization in  prepareIndex is fine.

Reply:
Yeah, I must admit I hesitated to add randomization on this (mostly because I wanted the tests to run on both index states every time). But this will go in master at some point so even with randomization it should catch any issues.